src/dataframe/dataframe.jl

```diff
@@ -218,6 +218,48 @@ function DataFrame(column_eltypes::AbstractVector{T}, cnames::AbstractVector{Sym
                      copycols=false)
 end
 
+"""
+    DataFrame!(args...; kwargs...)
+
+Equivalent to `DataFrame(args...; copycols=false, kwargs...)`
+or `DataFrame(args...; kwargs...)` if the specific `DataFrame` constructor
+does not accept `copycols` keyword argument.
+
+If `kwargs` contains the `copycols` keyword argument an error is thrown.
+
+### Examples
+
+```jldoctest
+julia> df1 = DataFrame(a=1:3)
+3×1 DataFrame
+│ Row │ a     │
+│     │ Int64 │
+├─────┼───────┤
+│ 1   │ 1     │
+│ 2   │ 2     │
+│ 3   │ 3     │
+
+julia> df2 = DataFrame!(df1)
+
+julia> df1.a === df2.a
+true
+"""
+function DataFrame!(args...; kwargs...)
+    if :copycols in keys(kwargs)
+        throw(ArgumentError("`copycols` keyword argument is not allowed"))
+    end
+    DataFrame(args...; copycols=false, kwargs...)
+end
+
+DataFrame!(columns::AbstractMatrix,
```

Comment:
I don't think we should support this. It's misleading since a copy is performed. A sign of this is that the docstring gets significantly more complex. Better not allow it for now, as always we can add them later if needed, but removing them is harder.

Reply:
Actually I had the same thought over the weekend. I will add errors in the cases we do not allow `copycols` kwarg (we could drop the methods and an error would be thrown anyway, but this way it will be more clear what is the reason of the problem).

There is one special case `DataFrame!()` which I still allowed (although technically it does not copy) as this is a special case calling `DataFrame(; kwargs...)` constructor.